nodes/node_battery_holder.lua

```diff
@@ -116,12 +99,8 @@ local def = {
 			return inv:add_item("batteries", stack)
 		end,
 		can_insert = function(pos, _, stack)
-			local meta = stack:get_metadata()
-			local md = minetest.deserialize(meta)
-			-- And specifically if they hold any charge
-			-- Disregard empty batteries, the player should know better
-			if md and md.charge > 0 then
-				meta = minetest.get_meta(pos)
+			if minetest.global_exists("technic") and technic.get_charge(stack) > 0 then
```

Comment:
Please keep the previous comment about `-- Disregard empty batteries, the player should know better` to justify the charge > 0 check.

Reply:
This function should be logically equivalent to `allow_metadata_inventory_put`. The player (automated or manual action) should be made aware that the battery is not usable as-is. Some feedback is needed for them, and rejecting the movement transaction is about the most simple approach to do that.
Anyway. I think this should be kept.